### PR TITLE
Docs: clean up Astro component scripts

### PR DESCRIPTION
**Pull request by DefinitelyMaybe**
This PR brings our Astro component scripts inline with [how astro does them](https://docs.astro.build/en/basics/astro-components/#the-component-script).

copied patterns were repeated. Anyone know where the `{ }` pattern came from?

```html
---
{
    'category': '@threlte/extras'
    'name': '<Align>'
}
---
```

vs

```html
---
category: '@threlte/extras'
name: '<Align>'
---
```

This PR will also make a move to astro+starlight easier. I ran into a wall of errors because of it when trying to make it happen myself.

### CODE REVIEW

File: apps/docs/src/content/reference/core/use-render.mdx [hunk 1]
```diff
@@ -22,8 +24,7 @@ Typically this hook is used to implement advanced rendering techniques such as p
 You may pass additional options to this hook. The property `order` is useful if you need to order the sequence of `useRender` callbacks across the component tree where callbacks are ordered from low to high.
 
 ```ts
-type ThrelteUseRenderOptions = {
-  order?: number
+type ThrelteUseRenderOptions = order?: number
```
Comment (michealparks) [inline apps/docs/src/content/reference/core/use-render.mdx:27]:
```suggestion
type ThrelteUseRenderOptions = {
  order?: number
}
```

Comment (DefinitelyMaybe) [reply apps/docs/src/content/reference/core/use-render.mdx]: good catch

**Comment by michealparks** [comment]
Should we try doing this in the next branch? Might be easier to get it pushed through there

**Comment by DefinitelyMaybe** [comment]
Hmm just noticed the new conflicts. I'll do one a merge main commit next week. Hopefully some approval comes, I still have hope 😅

File: apps/docs/src/content/reference/core/use-render.mdx [hunk 2]
```diff
@@ -25,6 +27,7 @@ You may pass additional options to this hook. The property `order` is useful if
 type ThrelteUseRenderOptions = {
   order?: number
 }
+}
```
Comment (michealparks) [inline apps/docs/src/content/reference/core/use-render.mdx:30]:
```suggestion
```

File: apps/docs/src/content/reference/theatre/project.mdx [hunk 3]
```diff
@@ -8,7 +8,7 @@ componentSignature:
   {
     props:
       [
-        {name: 'name', type: 'string', default: 'default', required: false},
+        {name: name, type: 'string', default: 'default', required: false},
```
Comment (michealparks) [inline apps/docs/src/content/reference/theatre/project.mdx:11]:
```suggestion
        {name: 'name', type: 'string', default: 'default', required: false},
```

**Pull request merged**